### PR TITLE
Verify that products in notebook category are presented

### PR DESCRIPTION
**Pull request by anastasia8755**

### CODE REVIEW

File: src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java [hunk 1]
```diff
@@ -17,4 +17,8 @@ public ProductTypePage chooseSubCategory(ProductCategoryAndSubCategory subCatego
         $x(format("//a[contains(@href,'%s')]", subCategory.getName())).click();
         return this;
     }
+
+    public boolean isThereMoreThanNProductsOnPagePresented(int n) {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java:21]: what's n?
re-do into getProducts() and do your comparison in the test

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/ss/ita/rozetka/ui/ProductsEnum/ProductCategoryAndSubCategory.java [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package com.ss.ita.rozetka.ui.ProductsEnum;
 
 public enum ProductCategoryAndSubCategory {
-    NOTEBOOKS("notebooks"),
+    NOTEBOOKS("notebooks/c80004/"),
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/ProductsEnum/ProductCategoryAndSubCategory.java:4]: investigate

File: src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java (hunk 1 above)
Comment (toleksyn) [reply src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java:21]: getProductsCount()

File: src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java [hunk 3]
```diff
@@ -17,4 +16,8 @@ public ProductTypePage chooseSubCategory(ProductCategoryAndSubCategory subCatego
         $x(format("//a[contains(@href,'%s')]", subCategory.getName())).click();
         return this;
     }
+
+    public int getProductsCount() {
+        return $$(".catalog-grid__cell").size();
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java:21]: this won't wait

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java (hunk 3 above)
Comment (toleksyn) [reply src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java:21]: $$x().shouldHaveSizeAtLeast(1).size()

File: src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java [hunk 4]
```diff
@@ -19,4 +20,12 @@ public ProductTypePage chooseSubCategory(ProductCategoryAndSubCategory subCatego
         $x(format("//a[contains(@href,'%s')]", subCategory.getName())).click();
         return this;
     }
+
+    @Step("ProductPage: get products count presented on page")
+    public int getProductsCount() {
+        return $$(".catalog-grid__cell")
+                .shouldHave(CollectionCondition
+                        .sizeGreaterThan(1))
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java:28]: format

File: src/test/com/ss/ita/rozetka/ProductsInCategoryArePresentedTest.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    @Test
+    public void verifyThatProductsArePresentedInNotebookCategory() {
+        ProductTypePage productTypePage = new HomePage()
+                .open()
+                .openProductCategoryPage(NOTEBOOKS_AND_COMPUTERS)
+                .openProductTypePage(NOTEBOOKS);
+
+        assertThat(productTypePage
+                .getProductsCount())
```
Comment (toleksyn) [inline src/test/com/ss/ita/rozetka/ProductsInCategoryArePresentedTest.java:22]: format

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/com/ss/ita/rozetka/ProductsInCategoryArePresentedTest.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import com.ss.ita.rozetka.ui.pages.HomePage;
+import com.ss.ita.rozetka.ui.pages.ProductTypePage;
+import org.testng.annotations.Test;
+import com.ss.ita.rozetka.ui.TestUtils.TestRunner;
+
+import static com.ss.ita.rozetka.ui.ProductsEnum.GeneralProductCategory.NOTEBOOKS_AND_COMPUTERS;
+import static com.ss.ita.rozetka.ui.ProductsEnum.ProductCategoryAndSubCategory.NOTEBOOKS;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ProductsInCategoryArePresentedTest extends TestRunner {
```
Comment (toleksyn) [inline src/test/com/ss/ita/rozetka/ProductsInCategoryArePresentedTest.java:12]: fix class name

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java [hunk 7]
```diff
@@ -24,4 +25,9 @@ public ProductTypePage chooseSubCategory(ProductCategoryAndSubCategory subCatego
     public Boolean isProductTypePageHeadingVisible() {
         return $x("//h1[@class = 'catalog-heading ng-star-inserted']").isDisplayed();
     }
+
+    @Step("ProductPage: get products count presented on page")
+    public int getProductsCount() {
+        return $$(".catalog-grid__cell").shouldHave(CollectionCondition.sizeGreaterThan(1)).size();
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java:31]: format

**Review by toleksyn — CHANGES_REQUESTED** [review]

**Review by dzhyv — APPROVED** [review]

**Review by NyxLex — APPROVED** [review]

**Review by NazarYakh — APPROVED** [review]

**Pull request merged**